### ISSUE TITLE
Form auto filled with someone else's info and contacts updated wrongly

### ISSUE THREAD

**Issue by rodrigocvb** [mautic/mautic#6075]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | X
| Feature request? | 
| Enhancement? | 

## Description:
A few months ago I [asked for support](https://www.mautic.org/topic/auto-fill-show-another-lead-information/#post-102828) but at that point the issue I faced didn't cause any trouble. Now, the same issue (from my computer I fill out a form, but it shows a lead's contact info instead) is messing things up. Even if I erase the lead information and fill it out with my own information, Mautic contacts section update (the last active column) both leads (me and him). And the worst: within the contact's record, it says that the lead filled out the form and shows nothing about my contact. It only work if I go incognito.
From the support ticket I had open, Peter explained me that it wasn't a bug, but an expected behavior. But now, with the record keeping the wrong information I supposed something could be wrong.

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.13.1
| PHP version | 7.0.27

### Steps to reproduce:
1. Open landing page with the form on Chrome (not incognito)
2. Form shows up with someone else's information (say Lead X)
3. I erase that information and fill it out with my own
4. Submit the form
5. Mautic says that Lead X filled out that form and only update 'last active' info of my record
 
### Log errors: 

It looks like there's no error.

**Comment by npracht** [comment]
Label: Bug

**Comment by npracht** [comment]
Label: Pending feedback

**Comment by escopecz** [comment]
Mautic forms doesn't display a contact information by default. I think that some step is missing. A link to the form would be great.

I don't understand the statement:

> It only work if I go incognito.

and then in the steps to reproduce there is stated that it should be done in an incognito browser. So does it work there or not?

**Comment by rodrigocvb** [comment]
The form is this one: https://mkt.fottorama.com.br/pt/software-impressao-instagram-hashtag

I meant that if I do the process of accessing this form in an incognito browser, it works: no lead info is shown and the correct Mautic record (mine) is updated if I fill out the form with mail email and personal info.

I have just accessed this address from Chrome (not in an incognito session) and the same lead's contact info shows up. I didn't performed any action. I went back to Mautic and it puts the lead on top of contacts list, as if he has just interacted somehow. But he didn't. It was me.
